core/src/main/java/io/elsci/assertreflectionequals/ErrorMessageBuilder.java

```diff
@@ -56,28 +58,62 @@ public ErrorMessageBuilder addShallowDiff(Object expected, Object actual) {
      * Builds part about different values in array. Example:
      * <pre>
      * --- Fields that differed ---
-     * bacteria.insect.id expected: 5
-     * bacteria.insect.id actual:   6</pre>
+     * bacteria.insect.id expected: [-5, 9, 17]
+     * bacteria.insect.id actual:   [-10, 8, 4]</pre>
      *
      * <pre>
      * intArray[0] expected: -5
      * intArray[0] actual:   -10
      * </pre>
      * @param expected expected array
      * @param actual actual array
-     * @param problemValues  contains mismatched values and corresponding index for array
+     * @param problemValues  contains mismatched values and their corresponding index in array
      */
     public ErrorMessageBuilder addArraysDiff(Object[] expected, Object[] actual, Map<Object, List<Object>> problemValues) {
-        this.addShallowDiff(Arrays.toString(expected), Arrays.toString(actual));
-
-        // For arrays only. We want to show which indices were different and values they had.
+        // Array may contain primitives or objects. We need to detect it before invoking 'addShallowDiff' method
+        // since if it's array with objects, we need firstly build an object array string
+        if (expected == null && actual.length == 0 || actual == null && expected.length == 0) {
+            this.addShallowDiff(Arrays.toString(expected), Arrays.toString(actual));
+        } else if (expected == null || expected.length == 0) {
```

Comment:
Since this repeats.. Maybe add a method `isBlank()`? Similar to what they have in commons-collections